core/citrus-api/src/main/java/com/consol/citrus/util/VariableExpressionIterator.java

```diff
@@ -0,0 +1,179 @@
+package com.consol.citrus.util;
+
+import com.consol.citrus.context.TestContext;
+import com.consol.citrus.exceptions.CitrusRuntimeException;
+import com.consol.citrus.util.VariableExpressionIterator.VariableSegment;
+import org.springframework.util.ReflectionUtils;
+import org.springframework.util.StringUtils;
+import java.lang.reflect.Array;
+import java.lang.reflect.Field;
+import java.util.Iterator;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class VariableExpressionIterator implements Iterator<VariableSegment> {
```

Comment:
this looks like a proper place for a short comment on how this iterator works, no? I'd be lost without knowing the issue and having access to the test class (`TestContextTest`).